### PR TITLE
Home Page with searching

### PR DESCRIPTION
**Pull request by cshinaver**
- [x] Create home page with search bar (preferably using a template for design's sake)
Searching functionality
- [x] Searching queries database book information 
- [x] Searching starts data scraper and dynamically adds search results to page (will require JS) 
- [ ] Implement caching

### CODE REVIEW

**Comment by cshinaver** [comment]
I created a pull request for this so we could have discussion here

**Comment by jhuseman** [comment]
I'm currently having a problem where the datascraper functions aren't working when called from api.py. Is it possible that request.args.get would be returning an integer instead of a string for the ISBN number?

**Comment by cshinaver** [comment]
Neither. It returns type unicode.

**Comment by cshinaver** [comment]
Also, the url would look better as `/api/

**Comment by jhuseman** [comment]
Should be ready to merge in (everything except caching). Please review.

File: bookfinder/api/scraper.py [hunk 1]
```diff
@@ -49,10 +49,23 @@ def __repr__(self):
...
 
+    def to_dict(self):
+        return {
+            'seller': self.seller,
+            'price': float(self.price),
+            'rental': self.is_rental,
+            'book_type': self.book_type,
+            'link': self.link,
+            'purchaseID': self.purchaseID
+            }
```
Comment (cshinaver) [inline bookfinder/api/scraper.py]: MOVE DAT BACK

File: bookfinder/api/scraper.py [hunk 2]
```diff
@@ -176,6 +199,8 @@ def get_page_for_Barnes_book_search(keyword):
         list_wrapper_item = soup.find('section', id='prodSummary')
         product_info = list_wrapper_item.find('li', class_='tab selected')
 
+        if product_info is None:
```
Comment (cshinaver) [inline bookfinder/api/scraper.py]: Add comment explaining early return

File: bookfinder/api/scraper.py [hunk 3]
```diff
@@ -185,7 +210,9 @@ def get_page_for_Barnes_book_search(keyword):
         new_PurchaseOption.purchaseID = ''
 
         item_price = product_info.find_all('a')[1]  # get price
-        new_PurchaseOption.price = item_price.get_text()
+        item_price_text = re.search(r"\$(.*)", item_price.get_text()).group(1)
+        print item_price_text
```
Comment (cshinaver) [inline bookfinder/api/scraper.py]: remove

File: bookfinder/api/scraper.py [hunk 4]
```diff
@@ -195,6 +222,8 @@ def get_page_for_Barnes_book_search(keyword):
         list_books.append(new_PurchaseOption)  # add purchased booko
 
         new_rental = soup.find('section', id='skuSelection')
+        if new_rental is None:
```
Comment (cshinaver) [inline bookfinder/api/scraper.py]: comment

File: bookfinder/api/scraper.py [hunk 5]
```diff
@@ -277,8 +270,10 @@ def extract_google_books_page_link_from_li(item):
             return a_item.attrs['href']
 
         content = get_google_books_search_page_for_isbn(isbn)
-        soup = BeautifulSoup(content)
+        soup = BeautifulSoup(content, 'html.parser')
         item = soup.find('li', class_='g')
+        if item is None:
```
Comment (cshinaver) [inline bookfinder/api/scraper.py]: comment

File: bookfinder/api/scraper.py [hunk 6]
```diff
@@ -420,4 +415,93 @@ def extract_google_books_price_list_from_redirect_link(redir_link):
         return option_list
 
     link = get_google_books_page_link_for_isbn(isbn)
+    if link is None:
+        print ("No results found at "
```
Comment (cshinaver) [inline bookfinder/api/scraper.py]: Remove print and add comment

File: bookfinder/api/scraper.py [hunk 7]
```diff
@@ -420,4 +415,93 @@ def extract_google_books_price_list_from_redirect_link(redir_link):
...
+        book.author = volume_info.get('authors')
+        image_links = volume_info.get(
+            'imageLinks'
+        )
+        if image_links:
+            book.thumbnail_link = image_links.get('thumbnail')
+
+        identifiers = volume_info['industryIdentifiers']
+        # if there is only 1 identifier, don't use this book option
+        if len(identifiers)<2:
```
Comment (cshinaver) [inline bookfinder/api/scraper.py]: Update to only use ISBN 13

File: bookfinder/api/views.py [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+        book_list.append(book.to_dict())
+    json_output = json.dumps(book_list, sort_keys=True, indent=4)
+    return json_output
+
+
+@app.route('/api/used_option_list/')
+def used_option_query():
+    def get_list_by_isbn(isbn):
+        def fill_list_by_bookid(option_list,book_id):
+            def add_to_list(list,option):
```
Comment (cshinaver) [inline bookfinder/api/views.py]: Change list variable name

File: bookfinder/api/views.py [hunk 9]
```diff
@@ -0,0 +1,106 @@
+from flask import render_template, request
+
+from bookfinder import app
+from bookfinder.db.connect import execute_sql_query
+from bookfinder.models.book import Book
+from bookfinder.models.purchasechoice import PurchaseChoice
+
+import json
+from scraper import get_book_object_list_for_book_title, get_amazon_books_for_keyword, get_Barnes_book_prices_for_isbn, get_google_books_for_isbn
```
Comment (cshinaver) [inline bookfinder/api/views.py]: Parenthisize this

File: bookfinder/api/views.py [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+                    option_list = add_to_list(option_list,option)
+            else:
+                option_list = add_to_list(option_list,query_return)
+            return option_list
+        
+        book_query = Book().get(isbn=isbn)
+        if book_query is None:#return blank list if this book is not yet in the database
+            option_list = []
+        elif isinstance(book_query,list):
+            print 'WARNING: There are multiple books with ISBN {isbn}'.format(isbn=isbn)
```
Comment (cshinaver) [inline bookfinder/api/views.py]: Remove print

File: bookfinder/api/views.py [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+    return json_output
+
+
+@app.route('/api/comparison_option_list/')
+def comparison_option_query():
+    isbn = request.args.get('isbn')
+    option_list = []
+    amazon_list = get_amazon_books_for_keyword(isbn)
+    if amazon_list is None:
+        print 'None returned from get_amazon_books_for_keyword'
```
Comment (cshinaver) [inline bookfinder/api/views.py]: Remove

File: bookfinder/api/views.py [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+    else:
+        print (
```
Comment (cshinaver) [inline bookfinder/api/views.py]: Remove

File: bookfinder/api/views.py [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+            " returned {count} result(s)".format(
+                isbn=isbn,
+                count=len(amazon_list)
+            )
+        )
+        for option in amazon_list:
+            option_list.append(option.to_dict())
+    barnes_list = get_Barnes_book_prices_for_isbn(isbn)
+    if barnes_list is None:
+        print 'None returned from get_Barnes_book_prices_for_isbn'
```
Comment (cshinaver) [inline bookfinder/api/views.py]: Remove

File: bookfinder/api/views.py [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+    else:
+        print (
```
Comment (cshinaver) [inline bookfinder/api/views.py]: Remove

File: bookfinder/api/views.py [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+            " returned {count} result(s)".format(
+                isbn=isbn,
+                count=len(barnes_list)
+            )
+        )
+        for option in barnes_list:
+            option_list.append(option.to_dict())
+    google_list = get_google_books_for_isbn(isbn)
+    if google_list is None:
+        print 'None returned from get_google_books_for_isbn'
```
Comment (cshinaver) [inline bookfinder/api/views.py]: Remove

File: bookfinder/api/views.py [hunk 16]
```diff
@@ -0,0 +1,106 @@
...
+    else:
+        print (
```
Comment (cshinaver) [inline bookfinder/api/views.py]: Remove

File: bookfinder/static/js/prices.js [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+	});
+}
+
+function fill_search(text) {
+	document.getElementById("head-search-bar_text").value = text;
+}
+
+function add_option_to_list(option, list_id) {
+	var new_row = document.createElement("tr");
+	//Seller:
```
Comment (cshinaver) [inline bookfinder/static/js/prices.js]: dedent

File: bookfinder/static/js/search.js [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+				add_book_to_list(book_list[i]);
+			}
+		} else { // doesn't seem to be working
+			document.getElementById("result-list").innerHTML = "Sorry. There are no Book results for the search query \"" + query + "\".";
+		}
+	});
+}
+
+function fill_search(text) {
+	//document.getElementById("head-search-bar_text").value = text;
```
Comment (cshinaver) [inline bookfinder/static/js/search.js]: Remove commented out text

File: bookfinder/static/js/search.js [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+	document.getElementById("main-search-bar_text").value = text;
+}
+
+function add_book_to_list(book) {
+	var new_entry = document.createElement("a");
+	//thumbnail:
+		var new_thumb_frame = document.createElement("div");
```
Comment (cshinaver) [inline bookfinder/static/js/search.js]: dedent plz

File: bookfinder/static/js/search.js [hunk 20]
```diff
@@ -0,0 +1,56 @@
...
+			var new_thumb = document.createElement("img");
+			new_thumb.setAttribute("src",book.Thumbnail_link);
+			//new_thumb.setAttribute("class","list-group-item-text");
```
Comment (cshinaver) [inline bookfinder/static/js/search.js]: remove commented code

File: bookfinder/templates/search/prices/index.html [hunk 21]
```diff
@@ -0,0 +1,53 @@
...
+			</tbody>
+		</table>
+	</div>
+{% endblock %}
+{% block footer %}
+    {{ super() }}
+	<script src="/static/js/get-file.js"></script>
+	<script src="/static/js/prices.js"></script>
+	<script>
+		//fill_search('{{ args.title }}');
```
Comment (cshinaver) [inline bookfinder/templates/search/prices/index.html]: remove commented code

**Comment by mperez14** [comment]
Approved

**Pull request merged**